### PR TITLE
Add the Hystrix stream implementation for OWIN

### PR DESCRIPTION
**Pull request by markvincze**
Add a new Stream implementation using OWIN, and extract the Configuration handling to a separate project so that it doesn't have to be duplicated.

### CODE REVIEW

**Comment by markvincze** [comment]
Thanks for issuing the PR, @klettier!
I took a quick look and looks good to me, I'll try to review it properly this week.

**Comment by klettier** [comment]
Thanks, I would like to change some little things like the order of the parameters in the extensions (in order to be consistent) and add the Owin sample project and tests with owin testing, do you think you could wait a bit?

**Comment by markvincze** [comment]
@klettier, yeah, sure, no problem ;)
Just ping when you're done.

**Comment by klettier** [comment]
## Changelog :

* Add Owin sample
* Fix parameters order for extension (improve consistency)
* Put extension into **Owin** namespace (trick to easily find it) 

## !! Breaking change !!

Note that due to the move of **WebConfiguration** into his proper project (**Hystrix.Dotnet.WebConfiguration**) the namespece of the object **HystrixConfigSection** have changed and must be modified accordingly in all **config file**.

PS: I will travel during 2 weeks from today so I couldn't contribute more during this period.

**Comment by klettier** [comment]
Any feedback on this?

**Comment by klettier** [comment]
I'm back, any news?

File: samples/Hystrix.Dotnet.Samples.Owin/Program.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+            container.Verify();
+
+            conf.DependencyResolver =
+                new SimpleInjector.Integration.WebApi.SimpleInjectorWebApiDependencyResolver(container);
+        }
+
+        static readonly Random rnd = new Random();
+
+        static Task<string> GetBreakerCommand(IHystrixCommand hystrixCommand, CancellationToken cancellationToken)
```
Comment (markvincze) [inline samples/Hystrix.Dotnet.Samples.Owin/Program.cs]: To me it doesn't feel like usual practice to set up an operation globally like this, I think it's usually more practical to set it up where we're actually doing it, in this case in the `HelloController`, could you move it there? Then the sample will be a bit simpler to understand too without the injection of a delegate.

File: src/Hystrix.Dotnet.Owin/Hystrix.Dotnet.Owin.csproj [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (markvincze) [inline src/Hystrix.Dotnet.Owin/Hystrix.Dotnet.Owin.csproj:1]: This csproj needs more info to properly do the Nuget release, like `PackageId`, `PackageProjectUrl`, etc. Can you add those properties? You can look at `Hystrix.Dotnet.AspNet.csproj` as an example.

**Comment by markvincze** [comment]
Please add the commands for building and testing the new Owin project to the `appveyor.yml`.

**Comment by markvincze** [comment]
Please adjust the namespace of `HystrixConfigSection` according to the breaking change everywhere we use it:

 - in the global README
 - in the README of the AspNet sample
 - in the `Web.config` of the AspNet sample

I think that's all the places.

**Comment by markvincze** [comment]
The last thing is bumping the version numbers. To make our lives easier we usually bump the version numbers of all the project released to Nuget (and keep them the same).
So you'll have to set the version number (in the `VersionPrefix` property) on the projects `Hystrix.Dotnet`, `Hystrix.Dotnet.AspNet`, `Hystrix.Dotnet.AspNetCore`, and the new `Hystrix.Dotnet.Owin`.

@JorritSalverda what do you think, do we have to bump to `2.0`, since the namespace change of the config section class is a breaking one, or is it enough to just make it `1.1.0`?

**Comment by JorritSalverda** [comment]
It's a breaking change, so we should definitely bump the major version. Otherwise we run the risk of breaking tons of Travix applications.

**Comment by klettier** [comment]
Hello,

Thank you for your feedback, I applied the changes accordingly and centralized **common** information in a **props** file for easy management.

I think It would be wise to provide packages in beta (2.0.0-beta1) and schedule the release in january ?

Regards,

kévin

**Comment by JorritSalverda** [comment]
Hey @klettier I still see this open PR, but I'm no longer maintainer of this repo. Can you remove me as reviewer of close the PR? Thanks.

**Comment by klettier** [comment]
@JorritSalverda I have no power here
foo

**Pull request closed without merging**